e2e/suite_test.go

```diff
@@ -142,13 +147,24 @@ func runScenario(ctx context.Context, t *testing.T, r *mrand.Rand, opts *scenari
 	// Only perform node readiness/pod-related checks when VMSS creation succeeded
 	if vmssSucceeded {
 		t.Log("vmss creation succeded, proceeding with node readiness and pod checks...")
-		if err = validateNodeHealth(ctx, t, opts.kube, vmssName); err != nil {
+		nodeName, err := validateNodeHealth(ctx, t, opts.kube, vmssName)
+		if err != nil {
 			t.Fatal(err)
 		}
 
+		if opts.nbc.AgentPoolProfile.WorkloadRuntime == datamodel.WasmWasi {
+			t.Log("wasm scenario: running wasm validation...")
+			if err := ensureWasmRuntimeClasses(ctx, opts.kube); err != nil {
+				t.Fatalf("unable to ensure wasm RuntimeClasses: %s", err)
+			}
+			if err := validateWasm(ctx, opts.kube, nodeName, string(privateKeyBytes)); err != nil {
```

Comment:
hmm...maybe we can abstract the validations sooner rather than later?

I'm guessing you didn't want to do this like the others because it's more kube than vmss layer?

Reply:
yeah exactly, though I do also see your point - could do something similar to what I did with the sysctl validation stuff, the main difference there is that the live VM validation is supposed to take place on the VM via exec + nsenter + ssh, while this validation is just supposed to take place on the pod via exec (pollExecOnPod vs. pollExecOnVM)

there's probably something we can do along the lines of splitting validators between VM and pod level